cel/README.md

```diff
@@ -37,231 +45,351 @@ kubectl apply --filename https://storage.cloud.google.com/tekton-releases-nightl
 
 ## Usage
 
-To evaluate a CEL expressions using `Custom Tasks`, we need to define a [`Run`](https://github.com/tektoncd/pipeline/blob/master/docs/runs.md)
-type with `apiVersion: cel.tekton.dev/v1alpha1` and `kind: CEL`. The `Run` takes the CEL expressions to be evaluated
-as `Parameters`. If executed successfully, the `Run` will produce the evaluation results as `Results` with names corresponding
-with the `Parameters` names. See the [examples](examples) folder for `CEL` `Custom Tasks` to run or use as samples. 
+To evaluate a CEL expressions using `Custom Tasks`, we need to define a [`Run`][run] type with 
+`apiVersion: custom.tekton.dev/v1alpha1` and `kind: CEL`. 
 
-### Configuring a `CEL` `Custom Task`
+The `Run` takes the CEL expressions to be evaluated through `expressions` field. The `Run` optionally takes CEL 
+environment variables through the `variables` field.
 
-The `CEL` `Custom Task` is defined in a `Run`, which supports the following fields:
+If executed successfully, the `Run` will produce the evaluation results as `Results` with names corresponding
+with the `Expressions`'s names. See the [examples](examples) folder for `CELs` to run or use as samples. 
 
-- [`apiVersion`][kubernetes-overview] - Specifies the API version, `tekton.dev/v1alpha1`
-- [`kind`][kubernetes-overview] - Identifies this resource object as a `Run` object
-- [`metadata`][kubernetes-overview] - Specifies the metadata that uniquely identifies the `Run`, such as a `name`
-- [`spec`][kubernetes-overview] - Specifies the configuration for the `Run`
-- [`ref`][kubernetes-overview] - Specifies the `CEL` `Custom Task`
-  - [`apiVersion`][kubernetes-overview] - Specifies the API version, `cel.tekton.dev/v1alpha1`
-  - [`kind`][kubernetes-overview] - Identifies this resource object as a `CEL` object
-- [`params`](#specifying-cel-expressions) - Specifies the CEL expressions to be evaluated as parameters
+### Configuring a `ELl`
+
+A `CEL` definition supports the following fields:
```

Comment:
im probably nitpicking but im wondering about naming the CRD `CEL` in that it feels strange to me to refer to `a CEL`

ultimately this seems to be a collection of CEL expressions, am i understanding that right? if so do we want to call it something like `CELExpressions`?

I think this also suggests it might be worthwhile to be more clear about defining exactly what the purpose and scope of this CRD is - if we make it really vague, there's no indication of what belongs in there and what doesn't.

these are probably more so considerations for later TEPs but if we can sort through some of it now it'll be less headache for users if we make changes

Reply:
> im probably nitpicking but im wondering about naming the CRD `CEL` in that it feels strange to me to refer to `a CEL`

given the pattern we have with `Task` --> `TaskRun`, `Pipelines` --> `PipelineRun`, so i thought it may make sense to call this `CEL` since we've been referring to the execution as `CELRun`

but i'm open to exploring other names for the CRD - maybe `CELEvaluation`?

> ultimately this seems to be a collection of CEL expressions, am i understanding that right? if so do we want to call it something like `CELExpressions`?

it's a collection of CEL expressions and environment variables

> I think this also suggests it might be worthwhile to be more clear about defining exactly what the purpose and scope of this CRD is - if we make it really vague, there's no indication of what belongs in there and what doesn't.
> 

the purpose of the CRD is to specify the CEL expressions to be evaluated given some (if any) environment variables

> these are probably more so considerations for later TEPs but if we can sort through some of it now it'll be less headache for users if we make changes

agreed that it makes sense to address them now